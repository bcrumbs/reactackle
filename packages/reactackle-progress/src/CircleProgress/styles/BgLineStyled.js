import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const style = ({ theme: themeFromProvider, customThickness }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    colourDefault,
    thickness,
  } = theme.reactackle.components.progress.circle.bgLine;

  return css`
    stroke: ${colourDefault};
    stroke-width: ${getValueString(customThickness || thickness)};
  `;
};

export const BgLineStyled = styled.circle`
  fill: none;
  stroke-dasharray: 0 158;
  transform-origin: center;
  ${style}
`;

BgLineStyled.displayName = 'BgLineStyled';
