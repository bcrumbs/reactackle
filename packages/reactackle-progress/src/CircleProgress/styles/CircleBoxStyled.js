import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const style = ({ theme: themeFromProvider, customDiameter }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    diameter,
  } = theme.reactackle.components.progress.circle;

  return css`
    width: ${getValueString(customDiameter || diameter)};
    height: ${getValueString(customDiameter || diameter)};
  `;
};

export const CircleBoxStyled = styled.svg`
  margin: auto;
  transform: rotate(-90deg);
  border-radius: 50%;
  display: block;
  max-width: 100%;
  max-height: 100%;
  ${style}
`;

CircleBoxStyled.displayName = 'CircleBoxStyled';
