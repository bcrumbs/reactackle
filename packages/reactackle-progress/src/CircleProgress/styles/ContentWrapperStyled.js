import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const style = ({ theme: themeFromProvider, customDiameter }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    diameter,
  } = theme.reactackle.components.progress.circle;

  return css`
    width: ${getValueString(customDiameter || diameter)};
    height: ${getValueString(customDiameter || diameter)};
  `;
};

export const ContentWrapperStyled = styled.div`
  margin: 0 auto;
  display: block;
  max-width: 100%;
  max-height: 100%;
  order: 2;
  ${style}
`;

ContentWrapperStyled.displayName = 'ContentWrapperStyled';
