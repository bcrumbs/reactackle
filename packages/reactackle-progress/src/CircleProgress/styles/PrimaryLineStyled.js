import styled, { css } from 'styled-components';
import { extractThemeOrDefault, getValueString } from 'reactackle-core';

const style = ({ theme: themeFromProvider, customThickness }) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const {
    colourDefault,
    thickness,
  } = theme.reactackle.components.progress.circle.mainLine;

  return css`
    stroke: ${colourDefault};
    stroke-width: ${getValueString(customThickness || thickness)};
  `;
};

export const PrimaryLineStyled = styled.circle`
  fill: none;
  stroke-dasharray: 0 158;
  transform-origin: center;
  ${style}
`;

PrimaryLineStyled.displayName = 'PrimaryLineStyled';
