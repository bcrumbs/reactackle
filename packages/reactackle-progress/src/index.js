import { registerDefaultComponentTheme } from 'reactackle-core';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('progress', componentTheme);

export { default as CircleProgress } from './CircleProgress/CircleProgress';
export * from './LinearProgress/LinearProgress';
export * from './Preloader/Preloader';
