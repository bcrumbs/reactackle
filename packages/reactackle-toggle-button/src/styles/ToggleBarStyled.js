import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import {
  extractThemeOrDefault,
  getValueString,
  transition,
  boxShadow,
} from 'reactackle-core';

const propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  /**
   * Define component theme config
   * See https://github.com/styled-components/styled-components/blob/master/docs/theming.md
   * for more information
   */
  theme: PropTypes.object,
};

const defaultProps = {
  checked: false,
  disabled: false,
};

/** PROP RECEIVERS */
const barBase = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const { width, height } = theme.reactackle.components.toggleButton.toggleBar;

  const thumbSize = theme.reactackle.components.toggleButton.toggleThumb.size,
    offsetTop = `calc((${getValueString(thumbSize)} - ${getValueString(
      height,
    )}) / 2)`;

  return `
    width: ${getValueString(width)};
    height: ${getValueString(height)};
    border-radius: ${getValueString(height)};
    top: ${offsetTop};
  `;
};

const barStyle = ({ theme: themeFromProvider, checked, disabled }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const componentPath =
    theme.reactackle.components.toggleButton.toggleBar.state;
  let source = null;

  if (!disabled)
    source = checked ? componentPath.checked : componentPath.unchecked;
  else source = componentPath.disabled;

  const backgroundColor = source.backgroundColor.toString();

  return `
    background-color: ${backgroundColor};
  `;
};

const thumbBase = ({ theme: themeFromProvider }) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const {
    size,
    borderRadius,
  } = theme.reactackle.components.toggleButton.toggleThumb;

  const barHeight = theme.reactackle.components.toggleButton.toggleBar.height;

  const offsetTop = `calc((${getValueString(barHeight)} - ${getValueString(
    size,
  )}) / 2)`;

  return css`
    width: ${getValueString(size)};
    height: ${getValueString(size)};
    top: ${offsetTop};
    border-radius: ${getValueString(borderRadius)};
  `;
};

const thumbStyle = ({
  theme: themeFromProvider,
  checked,
  disabled,
  hover,
  focus,
}) => {
  const theme = extractThemeOrDefault(themeFromProvider);
  const componentPath =
    theme.reactackle.components.toggleButton.toggleThumb.state;
  let source = null;

  if (!disabled)
    source = checked ? componentPath.checked : componentPath.unchecked;
  else source = componentPath.disabled;

  const { backgroundColor } = source;

  return css`
    background-color: ${backgroundColor};
    ${!disabled && boxShadow()};
    ${!disabled && (hover || focus) && boxShadow(2)};
  `;
};

const thumbPosition = ({
  checked,
  disabled,
  hover,
  focus,
  theme: themeFromProvider,
}) => {
  const theme = extractThemeOrDefault(themeFromProvider);

  const hoverOffset =
    theme.reactackle.components.toggleButton.toggleThumb.hoverOffset;

  const offset = (hover || focus) && !disabled
                 ? getValueString(hoverOffset) : 0,
    left = checked ? 'auto' : offset,
    right = checked ? offset : 'auto';

  return `
    left: ${left};
    right: ${right};
  `;
};

/** STYLES */
export const ToggleBarStyled = styled.label`
  display: block;
  position: absolute;
  left: 0;
  will-change: transform, box-shadow;
  ${barBase};
  ${barStyle};
  ${transition('background-color')};

  &::before {
    content: '';
    position: absolute;
    ${thumbBase};
    ${thumbPosition};
    ${thumbStyle};
    ${transition('left, box-shadow')};
  }
`;

ToggleBarStyled.propTypes = propTypes;
ToggleBarStyled.defaultProps = defaultProps;
ToggleBarStyled.displayName = 'ToggleBarStyled';
