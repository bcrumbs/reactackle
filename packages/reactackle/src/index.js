export * from 'reactackle-app';
export * from 'reactackle-autoposition';
export * from 'reactackle-button';
export * from 'reactackle-card';
export * from 'reactackle-checkbox';
export * from 'reactackle-core';
export * from 'reactackle-dialog';
export * from 'reactackle-form';
export * from 'reactackle-grid';
export * from 'reactackle-header';
export * from 'reactackle-icon';
export * from 'reactackle-input-autocomplete';
export * from 'reactackle-radio-group';
export * from 'reactackle-selectbox';
export * from 'reactackle-sidebar';
export * from 'reactackle-tabs';
export * from 'reactackle-text-field';
export * from 'reactackle-toggle-button';
export * from 'reactackle-tooltip';
export * from 'reactackle-tooltip-icon';
export * from 'reactackle-progress';
