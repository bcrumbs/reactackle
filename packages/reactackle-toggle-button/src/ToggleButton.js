'use strict';

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { TooltipIcon } from 'reactackle-tooltip-icon';
import { noop, isUndef, registerDefaultComponentTheme } from 'reactackle-core';

import { ToggleStyled } from './styles/ToggleStyled';
import { ToggleBarBoxStyled } from './styles/ToggleBarBoxStyled';
import { ToggleBarStyled } from './styles/ToggleBarStyled';
import { ToggleInputStyled } from './styles/ToggleInputStyled';
import { ToggleLabelStyled } from './styles/ToggleLabelStyled';
import { ToggleLabelTextBoxStyled } from './styles/ToggleLabelTextBoxStyled';
import { ToggleLabelTextStyled } from './styles/ToggleLabelTextStyled';
import { ToggleTooltipBoxStyled } from './styles/ToggleTooltipBoxStyled';
import componentTheme from './styles/theme';

registerDefaultComponentTheme('toggleButton', componentTheme);

const propTypes = {
  /**
   * Determines whether ToggleButton is checked
   */
  checked: PropTypes.bool,
  /**
   * Determines whether ToggleButton is checked by default
   */
  defaultChecked: PropTypes.bool,
  /**
   * Determines whether ToggleButton is disabled
   */
  disabled: PropTypes.bool,
  /**
   * Determines ToggleButton label
   */
  label: PropTypes.string,
  /**
   * Determines ToggleButton tooltip
   */
  tooltip: PropTypes.string,
  /**
   * Determines ToggleButton label position
   */
  labelPosition: PropTypes.oneOf(['left', 'right']),
  /**
   * Specify function to call on ToggleButton state change
   */
  onChange: PropTypes.func,
};
const defaultProps = {
  checked: void 0,
  defaultChecked: false,
  disabled: false,
  label: '',
  tooltip: '',
  labelPosition: 'left',
  onChange: noop,
};

let nextId = 0;

export default class ToggleButton extends Component {
  constructor(props) {
    super(props);

    this._inputId = `toggle-${nextId++}`;

    this.state = {
      checked: isUndef(props.checked) ? props.defaultChecked : props.checked,
      hovered: false,
      focused: false,
    };

    this._handleChange = this._handleChange.bind(this);
    this._handleMouseEnter = this._handleMouseEnter.bind(this);
    this._handleMouseLeave = this._handleMouseLeave.bind(this);
    this._handleFocus = this._handleFocus.bind(this);
    this._handleBlur = this._handleBlur.bind(this);
  }

  _handleChange(event) {
    const checked = event.target.checked;
    this.props.onChange({
      value: checked,
    });

    if (isUndef(this.props.checked)) this.setState({ checked });
  }

  _handleMouseEnter() {
    this.setState({
      hovered: true,
    });
  }

  _handleMouseLeave() {
    this.setState({
      hovered: false,
    });
  }

  _handleFocus() {
    this.setState({ focused: true });
  }

  _handleBlur() {
    this.setState({ focused: false });
  }

  _renderTooltip() {
    if (!this.props.tooltip) return null;

    return (
      <ToggleTooltipBoxStyled>
        <TooltipIcon text={this.props.tooltip} />
      </ToggleTooltipBoxStyled>
    );
  }

  _renderLabel() {
    if (!this.props.label) return null;

    const tooltip = this._renderTooltip(),
      checked = isUndef(this.props.checked)
        ? this.state.checked
        : this.props.checked;

    return (
      <ToggleLabelTextBoxStyled
        disabled={this.props.disabled}
        checked={checked}
        hasTooltip={this.props.tooltip}
        labelPosition={this.props.labelPosition}
      >
        <ToggleLabelTextStyled hasTooltip={this.props.tooltip}>
          {this.props.label}
        </ToggleLabelTextStyled>
        {tooltip}
      </ToggleLabelTextBoxStyled>
    );
  }

  render() {
    const checked = isUndef(this.props.checked)
      ? this.state.checked
      : this.props.checked;

    const label = this._renderLabel();

    return (
      <ToggleStyled
        onMouseEnter={this._handleMouseEnter}
        onMouseLeave={this._handleMouseLeave}
        onFocus={this._handleFocus}
        onBlur={this._handleBlur}
        disabled={this.props.disabled}
      >
        <ToggleLabelStyled
          htmlFor={this._inputId}
          tabIndex="0"
          disabled={this.props.disabled}
        >
          <ToggleBarBoxStyled
            labelPosition={this.props.labelPosition}
            disabled={this.props.disabled}
          >
            <ToggleBarStyled
              htmlFor={this._inputId}
              checked={checked}
              disabled={this.props.disabled}
              hover={this.state.hovered}
              focus={this.state.focused}
            />
            <ToggleInputStyled
              id={this._inputId}
              type="checkbox"
              disabled={this.props.disabled}
              checked={checked}
              onChange={this._handleChange}
              tabIndex={0}
            />
          </ToggleBarBoxStyled>
          {label}
        </ToggleLabelStyled>
      </ToggleStyled>
    );
  }
}

ToggleButton.propTypes = propTypes;
ToggleButton.defaultProps = defaultProps;
ToggleButton.displayName = 'ToggleButton';
