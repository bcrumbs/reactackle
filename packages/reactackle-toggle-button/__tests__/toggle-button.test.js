'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import { findByType } from 'reactackle-test-utils';
import { ToggleButton } from '../src';
import { ToggleStyled } from '../src/styles/ToggleStyled';

jest.mock('react-dom');

describe('<ToggleButton/>', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <ToggleButton />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if disabled', () => {
    const tree = renderer.create(
      <ToggleButton disabled />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if checked', () => {
    const tree = renderer.create(
      <ToggleButton checked />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if checked and disabled', () => {
    const tree = renderer.create(
      <ToggleButton checked disabled />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip and label', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with label', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label, labelPosition=right', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" labelPosition="right" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label and checked', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" checked />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label and disabled', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" disabled />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltip, label, disabled and checked', () => {
    const tree = renderer.create(
      <ToggleButton label="Foo" tooltip="Bar" disabled checked />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly if hovered', () => {
    const component = renderer.create(
      <ToggleButton />,
    );

    const toggle = findByType(component, ToggleStyled);

    expect(component.toJSON()).toMatchSnapshot();
    toggle.props.onMouseEnter();
    expect(component.toJSON()).toMatchSnapshot();
    toggle.props.onMouseLeave();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if focused', () => {
    const component = renderer.create(
      <ToggleButton />,
    );
    const toggle = findByType(component, ToggleStyled);

    expect(component.toJSON()).toMatchSnapshot();
    
    toggle.props.onFocus();
    expect(component.toJSON()).toMatchSnapshot();
    
    toggle.props.onBlur();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly if handles input change and call onChange', () => {
    const mockFn = jest.fn(),
      component = renderer.create(
        <ToggleButton onChange={mockFn} />,
      );
    findByType(component, 'input').props.onChange({
      target: {
        checked: true,
      },
    });

    expect(mockFn).toHaveBeenCalledWith({ value: true });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
