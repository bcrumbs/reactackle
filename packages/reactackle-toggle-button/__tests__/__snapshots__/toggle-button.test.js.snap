// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton/> renders correctly if checked 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-2"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB jXzuUU"
        disabled={false}
        htmlFor="toggle-2"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-2"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if checked and disabled 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-3"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB ihyzJK"
        disabled={true}
        htmlFor="toggle-3"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-3"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if disabled 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-1"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB haaxXZ"
        disabled={true}
        htmlFor="toggle-1"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-1"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if handles input change and call onChange 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-11"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB jXzuUU"
        disabled={false}
        htmlFor="toggle-11"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-11"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if hovered 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-10"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-10"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-10"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if hovered 2`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-10"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB gTkPlS"
        disabled={false}
        htmlFor="toggle-10"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-10"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly if hovered 3`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-10"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-10"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-10"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with default props 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-0"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-0"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-0"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with label 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-5"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-5"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-5"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn XtjXC"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni jSzCbs"
      >
        Foo
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip and label 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-4"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-4"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-4"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn XtjXC"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={1}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label and checked 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-7"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR heeuOl"
      disabled={false}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB jXzuUU"
        disabled={false}
        htmlFor="toggle-7"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-7"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={true}
      className="ToggleLabelTextBoxStyled-epGXVn XtjXC"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={1}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label and disabled 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-8"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB haaxXZ"
        disabled={true}
        htmlFor="toggle-8"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-8"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn lnKhrU"
      disabled={true}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={1}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label, disabled and checked 1`] = `
<div
  className="ToggleStyled-hpSctc hKhcQp"
  disabled={true}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz NOqCm"
    disabled={true}
    htmlFor="toggle-9"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR fHkSgk"
      disabled={true}
    >
      <label
        checked={true}
        className="ToggleBarStyled-kcTwBB ihyzJK"
        disabled={true}
        htmlFor="toggle-9"
      />
      <input
        checked={true}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={true}
        id="toggle-9"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={true}
      className="ToggleLabelTextBoxStyled-epGXVn lnKhrU"
      disabled={true}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={1}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;

exports[`<ToggleButton/> renders correctly with tooltip, label, labelPosition=right 1`] = `
<div
  className="ToggleStyled-hpSctc cNwmaA"
  disabled={false}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <label
    className="ToggleLabelStyled-iqJNYz cCwlqu"
    disabled={false}
    htmlFor="toggle-6"
    tabIndex="0"
  >
    <div
      className="ToggleBarBoxStyled-cBeZVR gapsjO"
      disabled={false}
    >
      <label
        checked={false}
        className="ToggleBarStyled-kcTwBB aFLLg"
        disabled={false}
        htmlFor="toggle-6"
      />
      <input
        checked={false}
        className="ToggleInputStyled-cniALy dfMZHO"
        disabled={false}
        id="toggle-6"
        onChange={[Function]}
        tabIndex={0}
        type="checkbox"
      />
    </div>
    <span
      checked={false}
      className="ToggleLabelTextBoxStyled-epGXVn bvDycJ"
      disabled={false}
    >
      <span
        className="ToggleLabelTextStyled-geBoni evgnHk"
      >
        Foo
      </span>
      <span
        className="ToggleTooltipBoxStyled-ehykpg jWsDJG"
      >
        <div
          className="TooltipIconStyled-egvEre kUiAus"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={1}
        >
          <span>
            ?
          </span>
        </div>
      </span>
    </span>
  </label>
</div>
`;
